### PR TITLE
Add option to set namespace to apply resources

### PR DESCRIPTION
**Pull request by pbstriker38**

### CODE REVIEW

**Review by askdesigners — COMMENTED** [review]
I'm no super expert but it looks pretty ok to me.

File: action.yaml [hunk 1]
```diff
@@ -16,9 +16,23 @@ inputs:
...
+    required: false
+    default: 'default'
 runs:
   using: 'composite'
   steps:
+    - name: Get current namespace
+      id: get-original-namespace
+      run: |
+        current_namespace=$(kubectl config view --minify -o jsonpath="{..namespace}")
+        current_namespace=${current_namespace:-"default"}
```
[the remark this answers is not in the document]
Comment (pbstriker38) [reply action.yaml:30]: It's an `||=`. So, if current_namespace has no value, then use "default"

[the remark this answers is not in the document]
Comment (askdesigners) [reply action.yaml:30]: ah cool. Never seen that syntax before. So this basically lets you use a generic script everywhere and to set the namespace by configs?

**Review by askdesigners — APPROVED** [review]

File: action.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pbstriker38) [reply action.yaml:30]: It allows you to apply the same configuration to multiple namespaces in k8. Only caveat is that the yaml can not define the namespace field cause that will take precedence over `kubectl` namespace

**Pull request merged**